pkg/pgmodel/new_migrate.go

```diff
@@ -145,19 +146,40 @@ func Migrate(conn *pgx.Conn, appVersion VersionInfo, leaseLock *util.PgAdvisoryL
 		if err = mig.Migrate(appSemver); err != nil {
```

Comment:
Is it necessary for this to be inside the explicit transaction you're creating?

Reply:
It is not strictly necessary though it would be nice. Unfortunately, the Migrate function starts its own explicit transactions somewhere in its path. So, it's not as straightforward to wrap it as I had hoped. I really don't want to touch that old code unless i absolutely have to.

Reply:
The Migrate function is the old way of doing database migrations. It would be nice if it were inside the transaction, but it's not strictly necessary. It creates its own explicit transactions, so it's not as easy as just wrapping it, unfortunately. What I mainly want to prevent here is a user getting stuck at version 0.0.0 which will be prevented by wrapping the rest in a transaction.